pkg/workloads/reservation.go

```diff
@@ -1235,3 +1219,35 @@ func userCanSign(userTid int64, req workloads.SigningRequest, signatures []workl
 
 	return nil
 }
+
+func isAllFreeToUse(ctx context.Context, nodeIDs []string, db *mongo.Database) (bool, error) {
+	var freeNodes int64
+	// check if freeTFT is allowed to be used
+	// if all nodes are marked as free to use then FreeTFT is allowed
+	// otherwise it is not
+	count, err := (directory.NodeFilter{}).
+		WithNodeIDs(nodeIDs).
+		WithFreeToUse(true).
+		Count(ctx, db)
+	if err != nil {
+		return false, err
+	}
+	freeNodes += count
+
+	// also include the gateways belonging to the farm
+	count, err = (directory.GatewayFilter{}).
+		WithGWIDs(nodeIDs).
+		WithFreeToUse(true).
+		Count(ctx, db)
+	if err != nil {
+		return false, err
+	}
+	freeNodes += count
+
+	log.Info().
+		Int("requested_nodes", len(nodeIDs)).
+		Int64("free_nodes", freeNodes).
+		Msg("distribution of free nodes in capacity reservation")
+
+	return freeNodes > int64(len(nodeIDs)), nil
```

Comment:
Should it not be `>=` because this value can be equal as well?

Reply:
right